[Bug]: build fail not able to run the project in macos 1 warning generated. /Users/trentiumsolution/.pub-cache/hosted/pub.dev/geolocator_apple-2.2.7/macos/Classes/Handlers/GeolocationHandler.m:63:22: warning: unused variable 'locationManager' [-Wunused-variable]   CLLocationManager *locationManager = [self getLocationManager];                      ^ 1 warning generated. 1 error generated. ** BUILD FAILED **

### Please check the following before submitting a new issue.

- [ ] I have searched the [existing issues](https://github.com/baseflow/flutter-geolocator/issues).
- [X] I have carefully [read the documentation](https://github.com/Baseflow/flutter-geolocator/blob/main/geolocator/README.md) and verified I have added the required platform specific configuration.

### Please select affected platform(s)

- [ ] Android
- [ ] iOS
- [ ] Linux
- [X] macOS
- [ ] Web
- [ ] Windows

### Steps to reproduce

1 warning generated.
/Users/trentiumsolution/.pub-cache/hosted/pub.dev/geolocator_apple-2.2.7/macos/Classes/Handlers/GeolocationHandler.m:63:22: warning: unused variable 'locationManager' [-Wunused-variable]
  CLLocationManager *locationManager = [self getLocationManager];
                     ^
1 warning generated.
1 error generated.
** BUILD FAILED **

### Expected results

getting this error 
Launching lib/main.dart on macOS in debug mode...
Running pod install...
Building macOS application...
/Users/trentiumsolution/.pub-cache/hosted/pub.dev/geolocator_apple-2.2.7/macos/Classes/Utils/LocationMapper.m:36:43: warning: 'sourceInformation' is only available on macOS 12.0 or newer [-Wunguarded-availability-new]
        [locationMap setObject:@(location.sourceInformation.isSimulatedBySoftware) forKey: @"is_mocked"];
                                          ^~~~~~~~~~~~~~~~~
In module 'CoreLocation' imported from /Users/trentiumsolution/.pub-cache/hosted/pub.dev/geolocator_apple-2.2.7/macos/Classes/Utils/LocationMapper.h:9:
@property(readonly, nonatomic, nullable) CLLocationSourceInformation *sourceInformation API_AVAILABLE(ios(15.0), watchos(8.0), tvos(15.0), macos(12.0));
                                                                      ^
        [locationMap setObject:@(location.sourceInformation.isSimulatedBySoftware) forKey: @"is_mocked"];
                                          ^~~~~~~~~~~~~~~~~
1 warning generated.
/Users/trentiumsolution/.pub-cache/hosted/pub.dev/geolocator_apple-2.2.7/macos/Classes/Handlers/GeolocationHandler.m:63:22: warning: unused variable 'locationManager' [-Wunused-variable]
  CLLocationManager *locationManager = [self getLocationManager];
                     ^
1 warning generated.
1 error generated.
** BUILD FAILED **

Exception: Build process failed


### Actual results

not able to run the macos build

### Code sample

<details><summary>Code sample</summary>

```dart
[Paste your code here]
```
1 warning generated.
/Users/trentiumsolution/.pub-cache/hosted/pub.dev/geolocator_apple-2.2.7/macos/Classes/Handlers/GeolocationHandler.m:63:22: warning: unused variable 'locationManager' [-Wunused-variable]
  CLLocationManager *locationManager = [self getLocationManager];
                     ^
1 warning generated.
1 error generated.
** BUILD FAILED **

</details>


### Screenshots or video

<details>
<summary>Screenshots or video demonstration</summary>

[Upload media here]

</details>
1 warning generated.
/Users/trentiumsolution/.pub-cache/hosted/pub.dev/geolocator_apple-2.2.7/macos/Classes/Handlers/GeolocationHandler.m:63:22: warning: unused variable 'locationManager' [-Wunused-variable]
  CLLocationManager *locationManager = [self getLocationManager];
                     ^
1 warning generated.
1 error generated.
** BUILD FAILED **

### Version

  geolocator: ^10.0.1

### Flutter Doctor output

<details><summary>Doctor output</summary>

```console
[Paste your output here]
```
[✓] Flutter (Channel stable, 3.13.5, on macOS 12.1 21C52 darwin-x64, locale
    en-GB)
    • Flutter version 3.13.5 on channel stable at
      /Users/trentiumsolution/development/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 12fccda598 (3 days ago), 2023-09-19 13:56:11 -0700
    • Engine revision bd986c5ed2
    • Dart version 3.1.2
    • DevTools version 2.25.0

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.2)
    • Android SDK at /Users/trentiumsolution/Library/Android/sdk
    • Platform android-33, build-tools 33.0.2
    • Java binary at: /Applications/Android
      Studio.app/Contents/jbr/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build
      17.0.6+0-17.0.6b802.4-9586694)
    • All Android licenses accepted.

[!] Xcode - develop for iOS and macOS (Xcode 13.4.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 13F100
    ✗ Flutter requires Xcode 14 or higher.
      Download the latest version or update via the Mac App Store.
    • CocoaPods version 1.12.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2022.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build
      17.0.6+0-17.0.6b802.4-9586694)

[✓] VS Code (version 1.77.3)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension can be installed from:
      🔨 https://marketplace.visualstudio.com/items?itemName=Dart-Code.flutter

[✓] Connected device (2 available)
    • macOS (desktop) • macos  • darwin-x64     • macOS 12.1 21C52 darwin-x64
    • Chrome (web)    • chrome • web-javascript • Google Chrome 116.0.5845.187

[✓] Network resources
    • All expected network resources are available.
</details>

--- a/geolocator_apple/CHANGELOG.md
+++ b/geolocator_apple/CHANGELOG.md
@@ -1,3 +1,7 @@
+## 2.3.2
+
+* Fixes build error and warnings regarding unused variables and unavailable APIs on macOS.
+
 ## 2.3.1
 
 * Fixes a bug where invalid speed and speed accuracy readings where returned instead of ignored.

--- a/geolocator_apple/ios/Classes/Handlers/GeolocationHandler.m
+++ b/geolocator_apple/ios/Classes/Handlers/GeolocationHandler.m
@@ -60,11 +60,11 @@ double const kMaxLocationLifeTimeInSeconds = 5.0;
   self.errorHandler = errorHandler;
   self.currentLocationResultHandler = resultHandler;
     
-  CLLocationManager *locationManager = [self getLocationManager];
   BOOL showBackgroundLocationIndicator = NO;
   BOOL allowBackgroundLocationUpdates = NO;
   #if TARGET_OS_IOS
     if (self.isListeningForPositionUpdates) {
+      CLLocationManager *locationManager = [self getLocationManager];
       showBackgroundLocationIndicator = locationManager.showsBackgroundLocationIndicator;
       allowBackgroundLocationUpdates = locationManager.allowsBackgroundLocationUpdates;
     }

--- a/geolocator_apple/ios/Classes/Utils/LocationMapper.m
+++ b/geolocator_apple/ios/Classes/Utils/LocationMapper.m
@@ -59,7 +59,7 @@
     // - https://developer.apple.com/documentation/corelocation/cllocation/3524338-courseaccuracy?language=objc
     double heading = location.course;
     if (heading >= 0.0) {
-        if (@available(iOS 13.4, *)) {
+        if (@available(iOS 13.4, macOS 10.15.4, *)) {
             double headingAccuracy = location.courseAccuracy;
             if (headingAccuracy >= 0.0) {
                 [locationMap setObject:@(heading) forKey: @"heading"];
@@ -70,7 +70,7 @@
         }
     }
     
-    if(@available(iOS 15.0, *)) {
+    if(@available(iOS 15.0, macOS 12.0, *)) {
         [locationMap setObject:@(location.sourceInformation.isSimulatedBySoftware) forKey: @"is_mocked"];
     }
     

--- a/geolocator_apple/pubspec.yaml
+++ b/geolocator_apple/pubspec.yaml
@@ -2,7 +2,7 @@ name: geolocator_apple
 description: Geolocation Apple plugin for Flutter. This plugin provides the Apple implementation for the geolocator.
 repository: https://github.com/baseflow/flutter-geolocator/tree/main/geolocator_apple
 issue_tracker: https://github.com/baseflow/flutter-geolocator/issues?q=is%3Aissue+is%3Aopen
-version: 2.3.1
+version: 2.3.2
 
 environment:
   sdk: ">=2.15.0 <4.0.0"

